src/phaser/map/MapManager.js

```diff
@@ -9,6 +9,10 @@ class MapManager {
 
     return map;
   }
+
+  static changeMap(scene, sceneKey) {
+    scene.scene.start(sceneKey);
+  }
```

Comment:
```suggestion
  static changeMap(scene, currentSceneKey, nextSceneKey) {
    scene.scene.stop(currentSceneKey);
    scene.scene.start(nextSceneKey);
  }
```

위에서 코드리뷰 남긴거랑 이어지는 부분인데, `Map`을 바꿀때 기존의 `Scene`을 파괴한다면 위와 같은 코드가 될꺼 같아요!

Reply:
일단 phaser의 [Scene lifecycle](https://rexrainbow.github.io/phaser3-rex-notes/docs/site/scene/#flow-chart)을 보면 더 빠르게 이해할 수 있을 것 같습니다.

그리고 [`scene.scene.start`](https://photonstorm.github.io/phaser3-docs/Phaser.Scenes.ScenePlugin.html#start__anchor)에는 현재 `scene`을 `shutdown`시키는 이벤트가 포함 돼 있습니다. 또한 [`remove`](https://photonstorm.github.io/phaser3-docs/Phaser.Scenes.ScenePlugin.html#remove__anchor) 메서드는 `Scene Manager`로부터 `scene`을 완전히 제거합니다. `scene`을 완전히 제거하게 되면 다시 불러올 수 있는 방법에 대해서 찾아보지는 않았지만, `scene`인스턴스가 완전히 사라지게 됩니다. 그러면 다시 불러오는 작업이 필요해지고, 이는 코드의 복잡성으로 이어진다고 생각해서 `phasre`에서 제공하는 인터페이스만으로 해결했습니다.

> `phaser`는 모든 `scene`을 한번에 불러와서 인스턴스화 시키고 인스턴스만의 lifecycle을 제공하기 때문에 `scene.shutdown`과 같은 메서드에 대해서 생성자가 다시 불리지 않습니다.

Reply:
> 오오 그렇군요.. 흠 그러면 scene에 이벤트가 등록할 때마다 위와 같은 cleanup 작업을 해줘야 하겠네요. 이벤트 등록이 많아질때마다 cleanup 작업도 계속 걸어줘야 하구요.. cleanup 작업을 scene 가 할 수 없을까요? shutdown 될때마다 등록된 이벤트 리스너를 전부 삭제해줬으면 좋겠는데

그리고 위 방법은 제가 열심히 찾아봤지만, 찾을 수 없었습니다. 아마 제 생각으로는 진영님이 이렇게 event listener를 제거해주는 코드에 대해서 코드 스멜을 맡으신 것 같은데, 저 또한 이게 정상적인 처리인가 라는 생각을 했었습니다. 그런데 생각해보면 `event listener`를 통한 처리는 위임을 통한 처리라고 할 수 있을 텐데 "`Scene`입장에서 모든 등록된 `listner`를 제거해주는게 `scene`의 역할이 맞나?" 라는 생각을 해보면 저는 아니라고 생각했고, 그제서야 이게 정상적인 패턴처럼 보였습니다.

결국 자기가 등록한 이벤트는 자기가 처리하는 그런 동작이라고 보면 될 것 같습니다.